playlist.c

```diff
@@ -830,9 +853,10 @@ bool playlist_push(playlist_t *playlist,
    {
       struct playlist_entry tmp;
       const char *entry_path = playlist->entries[i].path;
-      bool equal_path        =
-         (string_is_empty(real_path) && string_is_empty(entry_path)) ||
-         playlist_path_equal(real_path, entry_path, fuzzy_archive_match);
+      const char* entry_relative_path = playlist->entries[i].relative_path;
+      bool equal_path =
+         (!string_is_empty(real_path) && !string_is_empty(entry_path) && playlist_path_equal(real_path, entry_path, fuzzy_archive_match)) ||
+         (!string_is_empty(relative_path) && !string_is_empty(entry_relative_path) && (strcmp(relative_path, entry_relative_path) == 0));
```

Comment:
Surely these `relative_path` path checks are redundant...? `real_path` and `entry_path` are all that should be needed here.